src/Development/Shake/Internal/Progress.hs

```diff
@@ -205,12 +205,26 @@ message input = liftA3 (,,) time perc debug
 --   while time left is calculated by scaling @remaining@ by the observed work rate in this build,
 --   roughly @done / time_elapsed@.
 progressDisplay :: Double -> (String -> IO ()) -> IO Progress -> IO ()
-progressDisplay sample disp prog = do
-    disp "Starting..." -- no useful info at this stage
+progressDisplay sample disp prog = progressRaw sample (disp . format) prog
+  where format Starting     = "Starting..."
+        format (Finished t) = "Finished in " ++ showDuration t
+        format (Executing p t secs perc done todo predicted) =
+            "Running for " ++ showDurationSecs t ++ " [" ++ show done ++ "/" ++ show todo ++ "]" ++
+            ", predicted " ++ predicted ++
+            maybe "" (", Failure! " ++) (isFailure p)
+
+data RawProgress
+  = Starting
+  | Finished Double
+  | Executing Progress Double Double Double Int Int String
```

Comment:
Running or Executing? I tend to think of executing as more "calling exec". I'm happy with whatever you prefer though.